src/semantic-router/pkg/extproc/reason_mode_selector.go

```diff
@@ -45,6 +47,25 @@ func (r *OpenAIRouter) getReasoningModeAndCategory(query string) (bool, string)
 	return false, categoryName
 }
 
+// getModelFamilyAndTemplateParam returns a normalized model family name and the template param to be used (if any)
+func getModelFamilyAndTemplateParam(model string) (string, string) {
+	lower := strings.ToLower(strings.TrimSpace(model))
+	if strings.Contains(lower, "qwen3") {
+		return "qwen3", "enable_thinking"
+	}
+	if strings.Contains(lower, "deepseek") || strings.Contains(lower, "ds") {
```

Comment:
`ds` is too broad to me. Maybe we should also add a config validation to detect if there are conflict or duplicate model names with `ds` in them but not actually deepseek..

Reply:
my setup modelname for deepseek-v3 is also ds-v3 😄